lib/matplotlib/backend_managers.py

```diff
@@ -74,6 +73,38 @@ def __init__(self, canvas):
         self.keypresslock = widgets.LockDraw()
         self.messagelock = widgets.LockDraw()
 
+    @property
+    def manager(self):
+        """FigureManager that owns this ToolManager"""
+        return self._manager
+
+    @property
+    def canvas(self):
+        """Canvas managed by FigureManager"""
+        return self._figure.canvas
+
+    @property
+    def figure(self):
+        """Figure that holds the canvas"""
+        return self._figure
+
+    def set_figure(self, figure):
```

Comment:
Ahh, that problem, I recall running into it before... I just dislike the asymmetry of ``a = foo`` and ``set_foo(a)``, rather than ``foo = a``, I find asymmetry makes the API more complicated to learn.